test/stdlib/tgmath.swift.gyb

```diff
@@ -273,6 +285,21 @@ MathTests.test("gradient_${T}") {
   expectEqualWithTolerance(5.0, fmaGrad.0, ulps: 16)
   expectEqualWithTolerance(4.0, fmaGrad.1, ulps: 16)
   expectEqualWithTolerance(1.0, fmaGrad.2, ulps: 16)
+  expectEqualWithTolerance(0.0, gradient(at: 2.0 as ${T}, in: { ceil($0) }), ulps: 16)
+  expectEqualWithTolerance(0.0, gradient(at: 2.0 as ${T}, in: { floor($0) }), ulps: 16)
+  expectEqualWithTolerance(0.0, gradient(at: 2.0 as ${T}, in: { round($0) }), ulps: 16)
+  expectEqualWithTolerance(0.0, gradient(at: 2.0 as ${T}, in: { trunc($0) }), ulps: 16)
+  for a in -10...10 {
+    let x = ${T}(a)
+    for b in -10...10 {
+      let y = ${T}(b)
+      guard b != 0 && remainder(x, y).sign == remainder(x + ${T}(0.001), y).sign &&
+        remainder(x, y).sign == remainder(x, y + ${T}(0.001)).sign
```

Comment:
I found out that for the combinations of `x` and `y` that don't satisfy this condition, computing the derivative for `remainder` using the definition of derivative would give unexpected results ( I guess due to the fact of it being discontinus ) hence I exclude them.